src/main/java/mandel/fallingsand/SandComponent.java

```diff
@@ -0,0 +1,32 @@
+package mandel.fallingsand;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.Random;
+
+public class SandComponent extends JComponent {
+
+    private final Sand sand;
+    private Random random;
+
+    public SandComponent(Sand sand) {
+        this.sand = sand;
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+
+        // draw the sand
+        sand.fall();
+        repaint();
+
+        for (int y = 0; y < sand.getHeight(); y++) {
+            for (int x = 0; x < sand.getWidth(); x++) {
+                if (sand.get(x, y) == 1) {
+                    g.fillRect(x * 2, y, 2, 2);
```

Comment:
Why not y*2 as well?

Reply:
It falls too far down and it can only be seen if I extend the screen. I did x * 2 so it would make it wider, otherwise it was only falling in a small rectangle area.